android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java

```diff
@@ -92,26 +91,20 @@ protected int getLayoutResource() {
     private void setupViewPager(final ViewPager viewPager) {
         adapter = new ScheduleViewPagerAdapter(getChildFragmentManager());
 
-        realmRepo.getEventDates()
-                .addChangeListener((eventDates, orderedCollectionChangeSet) -> {
-                    int eventDays = eventDates.size();
-                    for (int i = 0; i < eventDays; i++) {
-                        String date = eventDates.get(i).getDate();
-
-                        try {
-                            adapter.addFragment(new DayScheduleFragment(),
-                                    DateConverter.formatDay(date), date);
-                            adapter.notifyDataSetChanged();
-                        } catch (DateTimeParseException pe) {
-                            Timber.e(pe);
-                            Timber.e("Invalid date %s in database", date);
-                        }
-                    }
-                });
+        scheduleFragmentViewModel.getEventDatesString().observe(this, eventDatesString -> {
+            int eventDays = eventDatesString.size();
+            for (int i = 0; i < eventDays; i++) {
+                String date = eventDatesString.get(i).first;
+                String formattedDate = eventDatesString.get(i).second;
+                adapter.addFragment(new DayScheduleFragment(),
+                        formattedDate, date);
+                adapter.notifyDataSetChanged();
+            }
+        });
```

Comment:
Move the logic of calculation of the loop inside ViewModel, the view should simply receive `eventDatesString.get(i)`

Reply:
@iamareebjamal The ```adapter.addFragment``` method needs both the ```date``` as well as the ```formattedDate``` variable that's why i have to pass them both as a ```Pair```. Perhaps i should name the lambda parameter as ```eventDatesStringPair``` instead of ```eventDatesString```. Please explain a bit more about how i can move the logic of calculation to ViewModel.

Reply:
Move the loop in ViewModel and then in the for loop, create formattedDate and date, then create a pair and pass it here

So in view, it will only look like this:

```Java
scheduleFragmentViewModel.getEventDatesString().observe(datePair -> {
    adapter.addFragment(new DayScheduleFragment(),	datePair.first, datePair.second);
    adapter.notifyDataSetChanged();
});
```